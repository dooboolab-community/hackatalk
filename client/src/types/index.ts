import { StyleProp, TextStyle } from 'react-native';

import { SFC } from 'react';
import { User } from './graphql';

export enum MessageType {
  Message,
  Photo,
  File,
}

export interface Friend extends User {
  isFriend?: boolean;
  friendSince: Date;
}

interface IconProps {
  style?: StyleProp<TextStyle>;
  width?: number | string;
  height?: number | string;
  children?: never;
}

export type IconType = SFC<IconProps>;

export enum ThemeType {
  LIGHT = 'LIGHT',
  DARK = 'DARK',
}
