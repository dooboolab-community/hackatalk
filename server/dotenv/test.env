JWT_SECRET=dooboolab
DATABASE_URL="postgresql://test:test!@localhost:5432/postgres?schema=test"
