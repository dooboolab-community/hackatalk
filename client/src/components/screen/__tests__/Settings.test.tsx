import 'react-native';

import * as React from 'react';

import { AuthType, User } from '../../../types/graphql';
import {
  RenderAPI,
  act,
  cleanup,
  fireEvent,
  render,
} from '@testing-library/react-native';
import { createTestElement, createTestProps } from '../../../../test/testUtils';

import AuthContext from '../../../providers/AuthProvider';
import Settings from '../Settings';

let component: React.ReactElement;

function getEmptyAuthUserWithSignInType(signInType: AuthType): User {
  return {
    id: '',
    nickname: '',
    thumbURL: '',
    photoURL: '',
    statusMessage: '',
  };
}

function SettingTest(): React.ReactElement {
  const settingProps = createTestProps();

  return <Settings {...settingProps} />;
}

describe('[Setting] screen', () => {
  let testingLib: RenderAPI;

  beforeEach(() => {
    component = createTestElement(
      <SettingTest />,
      undefined,
      undefined,
      getEmptyAuthUserWithSignInType(AuthType.Email),
    );

    testingLib = render(component);

    const json = testingLib.toJSON();

    expect(json).toMatchSnapshot();
  });

  it('renders without crashing with Facebook auth type', () => {
    component = createTestElement(
      <SettingTest />,
      undefined,
      undefined,
      getEmptyAuthUserWithSignInType(AuthType.Facebook),
    );

    testingLib = render(component);

    const json = testingLib.toJSON();

    expect(json).toMatchSnapshot();
  });

  it('should renders without crashing with Google auth type', () => {
    component = createTestElement(
      <SettingTest />,
      undefined,
      undefined,
      getEmptyAuthUserWithSignInType(AuthType.Google),
    );

    testingLib = render(component);

    const json = testingLib.toJSON();

    expect(json).toMatchSnapshot();
  });

  it('should render without crashing with Apple auth type', () => {
    component = createTestElement(
      <SettingTest />,
      undefined,
      undefined,
      getEmptyAuthUserWithSignInType(AuthType.Apple),
    );

    testingLib = render(component);

    const json = testingLib.toJSON();

    expect(json).toMatchSnapshot();
  });

  describe('interactions', () => {
    beforeEach(() => {
      testingLib = render(component);
    });

    it('should simulate onPress login state item', async () => {
      const btn = testingLib.getByTestId('change-pw-item');

      fireEvent.press(btn);
    });

    it('should fireEvent when logout button pressed', () => {
      jest
        .spyOn(AuthContext, 'useAuthContext')
        .mockImplementation(() => ({
          state: {
            user: undefined,
          },
          setUser: jest.fn().mockReturnValue({
            id: 'userId',
            email: 'email@email.com',
            nickname: 'nickname',
            statusMessage: 'status',
          }),
        }));

      act(() => {
        fireEvent.press(testingLib.getByTestId('button-logout'));
      });
    });
  });

  afterAll(() => {
    cleanup();
  });
});
