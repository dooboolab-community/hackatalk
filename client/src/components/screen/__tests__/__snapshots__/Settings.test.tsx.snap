// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Setting] screen interactions should fireEvent when logout button pressed 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingTop": 10,
      },
    ]
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "data": Array [
            Object {
              "label": "Signed in with Email",
              "onPress": [Function],
              "testID": "change-pw-item",
            },
          ],
          "title": "Login information",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={
      Array [
        0,
      ]
    }
    testID="test-section-list"
    updateCellsBatchingPeriod={50}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#ffffff",
                "borderBottomColor": "#d3d8e8",
                "borderBottomWidth": 1,
                "height": 40,
                "justifyContent": "center",
                "marginLeft": 12,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#1E6EFA",
                  "marginLeft": 2,
                },
              ]
            }
          >
            Login information
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 52,
              "opacity": 1,
              "paddingLeft": 15,
              "paddingRight": 15,
              "width": "100%",
            }
          }
          testID="change-pw-item"
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#151A25",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "fontSize": 16,
                },
              ]
            }
          >
            Signed in with Email
          </Text>
          <Text />
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      />
    </View>
  </RCTScrollView>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "button": Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#1E6EFA",
          "borderWidth": 0.3,
          "height": 48,
          "width": "100%",
        },
        "opacity": 1,
        "root": Object {
          "paddingHorizontal": 20,
          "paddingVertical": 10,
        },
        "text": Object {
          "color": "#1E6EFA",
          "fontSize": 14,
          "fontWeight": "bold",
        },
      }
    }
    testID="button-logout"
  />
</RCTSafeAreaView>
`;

exports[`[Setting] screen interactions should simulate onPress login state item 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingTop": 10,
      },
    ]
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "data": Array [
            Object {
              "label": "Signed in with Email",
              "onPress": [Function],
              "testID": "change-pw-item",
            },
          ],
          "title": "Login information",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={
      Array [
        0,
      ]
    }
    testID="test-section-list"
    updateCellsBatchingPeriod={50}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#ffffff",
                "borderBottomColor": "#d3d8e8",
                "borderBottomWidth": 1,
                "height": 40,
                "justifyContent": "center",
                "marginLeft": 12,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#1E6EFA",
                  "marginLeft": 2,
                },
              ]
            }
          >
            Login information
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 52,
              "opacity": 1,
              "paddingLeft": 15,
              "paddingRight": 15,
              "width": "100%",
            }
          }
          testID="change-pw-item"
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#151A25",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "fontSize": 16,
                },
              ]
            }
          >
            Signed in with Email
          </Text>
          <Text />
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      />
    </View>
  </RCTScrollView>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "button": Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#1E6EFA",
          "borderWidth": 0.3,
          "height": 48,
          "width": "100%",
        },
        "opacity": 1,
        "root": Object {
          "paddingHorizontal": 20,
          "paddingVertical": 10,
        },
        "text": Object {
          "color": "#1E6EFA",
          "fontSize": 14,
          "fontWeight": "bold",
        },
      }
    }
    testID="button-logout"
  />
</RCTSafeAreaView>
`;

exports[`[Setting] screen renders without crashing with Facebook auth type 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingTop": 10,
      },
    ]
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "data": Array [
            Object {
              "label": "Signed in with Email",
              "onPress": [Function],
              "testID": "change-pw-item",
            },
          ],
          "title": "Login information",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={
      Array [
        0,
      ]
    }
    testID="test-section-list"
    updateCellsBatchingPeriod={50}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#ffffff",
                "borderBottomColor": "#d3d8e8",
                "borderBottomWidth": 1,
                "height": 40,
                "justifyContent": "center",
                "marginLeft": 12,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#1E6EFA",
                  "marginLeft": 2,
                },
              ]
            }
          >
            Login information
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 52,
              "opacity": 1,
              "paddingLeft": 15,
              "paddingRight": 15,
              "width": "100%",
            }
          }
          testID="change-pw-item"
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#151A25",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "fontSize": 16,
                },
              ]
            }
          >
            Signed in with Email
          </Text>
          <Text />
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      />
    </View>
  </RCTScrollView>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "button": Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#1E6EFA",
          "borderWidth": 0.3,
          "height": 48,
          "width": "100%",
        },
        "opacity": 1,
        "root": Object {
          "paddingHorizontal": 20,
          "paddingVertical": 10,
        },
        "text": Object {
          "color": "#1E6EFA",
          "fontSize": 14,
          "fontWeight": "bold",
        },
      }
    }
    testID="button-logout"
  />
</RCTSafeAreaView>
`;

exports[`[Setting] screen renders without crashing with Facebook auth type 2`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingTop": 10,
      },
    ]
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "data": Array [
            Object {
              "label": "Signed in with Email",
              "onPress": [Function],
              "testID": "change-pw-item",
            },
          ],
          "title": "Login information",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={
      Array [
        0,
      ]
    }
    testID="test-section-list"
    updateCellsBatchingPeriod={50}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#ffffff",
                "borderBottomColor": "#d3d8e8",
                "borderBottomWidth": 1,
                "height": 40,
                "justifyContent": "center",
                "marginLeft": 12,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#1E6EFA",
                  "marginLeft": 2,
                },
              ]
            }
          >
            Login information
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 52,
              "opacity": 1,
              "paddingLeft": 15,
              "paddingRight": 15,
              "width": "100%",
            }
          }
          testID="change-pw-item"
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#151A25",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "fontSize": 16,
                },
              ]
            }
          >
            Signed in with Email
          </Text>
          <Text />
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      />
    </View>
  </RCTScrollView>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "button": Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#1E6EFA",
          "borderWidth": 0.3,
          "height": 48,
          "width": "100%",
        },
        "opacity": 1,
        "root": Object {
          "paddingHorizontal": 20,
          "paddingVertical": 10,
        },
        "text": Object {
          "color": "#1E6EFA",
          "fontSize": 14,
          "fontWeight": "bold",
        },
      }
    }
    testID="button-logout"
  />
</RCTSafeAreaView>
`;

exports[`[Setting] screen should render without crashing with Apple auth type 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingTop": 10,
      },
    ]
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "data": Array [
            Object {
              "label": "Signed in with Email",
              "onPress": [Function],
              "testID": "change-pw-item",
            },
          ],
          "title": "Login information",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={
      Array [
        0,
      ]
    }
    testID="test-section-list"
    updateCellsBatchingPeriod={50}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#ffffff",
                "borderBottomColor": "#d3d8e8",
                "borderBottomWidth": 1,
                "height": 40,
                "justifyContent": "center",
                "marginLeft": 12,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#1E6EFA",
                  "marginLeft": 2,
                },
              ]
            }
          >
            Login information
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 52,
              "opacity": 1,
              "paddingLeft": 15,
              "paddingRight": 15,
              "width": "100%",
            }
          }
          testID="change-pw-item"
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#151A25",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "fontSize": 16,
                },
              ]
            }
          >
            Signed in with Email
          </Text>
          <Text />
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      />
    </View>
  </RCTScrollView>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "button": Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#1E6EFA",
          "borderWidth": 0.3,
          "height": 48,
          "width": "100%",
        },
        "opacity": 1,
        "root": Object {
          "paddingHorizontal": 20,
          "paddingVertical": 10,
        },
        "text": Object {
          "color": "#1E6EFA",
          "fontSize": 14,
          "fontWeight": "bold",
        },
      }
    }
    testID="button-logout"
  />
</RCTSafeAreaView>
`;

exports[`[Setting] screen should render without crashing with Apple auth type 2`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingTop": 10,
      },
    ]
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "data": Array [
            Object {
              "label": "Signed in with Email",
              "onPress": [Function],
              "testID": "change-pw-item",
            },
          ],
          "title": "Login information",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={
      Array [
        0,
      ]
    }
    testID="test-section-list"
    updateCellsBatchingPeriod={50}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#ffffff",
                "borderBottomColor": "#d3d8e8",
                "borderBottomWidth": 1,
                "height": 40,
                "justifyContent": "center",
                "marginLeft": 12,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#1E6EFA",
                  "marginLeft": 2,
                },
              ]
            }
          >
            Login information
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 52,
              "opacity": 1,
              "paddingLeft": 15,
              "paddingRight": 15,
              "width": "100%",
            }
          }
          testID="change-pw-item"
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#151A25",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "fontSize": 16,
                },
              ]
            }
          >
            Signed in with Email
          </Text>
          <Text />
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      />
    </View>
  </RCTScrollView>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "button": Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#1E6EFA",
          "borderWidth": 0.3,
          "height": 48,
          "width": "100%",
        },
        "opacity": 1,
        "root": Object {
          "paddingHorizontal": 20,
          "paddingVertical": 10,
        },
        "text": Object {
          "color": "#1E6EFA",
          "fontSize": 14,
          "fontWeight": "bold",
        },
      }
    }
    testID="button-logout"
  />
</RCTSafeAreaView>
`;

exports[`[Setting] screen should renders without crashing with Google auth type 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingTop": 10,
      },
    ]
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "data": Array [
            Object {
              "label": "Signed in with Email",
              "onPress": [Function],
              "testID": "change-pw-item",
            },
          ],
          "title": "Login information",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={
      Array [
        0,
      ]
    }
    testID="test-section-list"
    updateCellsBatchingPeriod={50}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#ffffff",
                "borderBottomColor": "#d3d8e8",
                "borderBottomWidth": 1,
                "height": 40,
                "justifyContent": "center",
                "marginLeft": 12,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#1E6EFA",
                  "marginLeft": 2,
                },
              ]
            }
          >
            Login information
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 52,
              "opacity": 1,
              "paddingLeft": 15,
              "paddingRight": 15,
              "width": "100%",
            }
          }
          testID="change-pw-item"
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#151A25",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "fontSize": 16,
                },
              ]
            }
          >
            Signed in with Email
          </Text>
          <Text />
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      />
    </View>
  </RCTScrollView>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "button": Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#1E6EFA",
          "borderWidth": 0.3,
          "height": 48,
          "width": "100%",
        },
        "opacity": 1,
        "root": Object {
          "paddingHorizontal": 20,
          "paddingVertical": 10,
        },
        "text": Object {
          "color": "#1E6EFA",
          "fontSize": 14,
          "fontWeight": "bold",
        },
      }
    }
    testID="button-logout"
  />
</RCTSafeAreaView>
`;

exports[`[Setting] screen should renders without crashing with Google auth type 2`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingTop": 10,
      },
    ]
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "data": Array [
            Object {
              "label": "Signed in with Email",
              "onPress": [Function],
              "testID": "change-pw-item",
            },
          ],
          "title": "Login information",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={
      Array [
        0,
      ]
    }
    testID="test-section-list"
    updateCellsBatchingPeriod={50}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#ffffff",
                "borderBottomColor": "#d3d8e8",
                "borderBottomWidth": 1,
                "height": 40,
                "justifyContent": "center",
                "marginLeft": 12,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#1E6EFA",
                  "marginLeft": 2,
                },
              ]
            }
          >
            Login information
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 52,
              "opacity": 1,
              "paddingLeft": 15,
              "paddingRight": 15,
              "width": "100%",
            }
          }
          testID="change-pw-item"
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#151A25",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "fontSize": 16,
                },
              ]
            }
          >
            Signed in with Email
          </Text>
          <Text />
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      />
    </View>
  </RCTScrollView>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "button": Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#1E6EFA",
          "borderWidth": 0.3,
          "height": 48,
          "width": "100%",
        },
        "opacity": 1,
        "root": Object {
          "paddingHorizontal": 20,
          "paddingVertical": 10,
        },
        "text": Object {
          "color": "#1E6EFA",
          "fontSize": 14,
          "fontWeight": "bold",
        },
      }
    }
    testID="button-logout"
  />
</RCTSafeAreaView>
`;
